packages/protocol/src/utils/type-helper.ts

```diff
@@ -9,3 +9,9 @@ export type ValuesAsUnionDeep<T> = {
 declare const brand: unique symbol;
 
 export type Brand<T, TBrand> = T & { [brand]: TBrand };
+
+export type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never;
+
+export type Prettify<T> = {
+  [K in keyof T]: T[K];
+} & {};
```

Comment:
typehelper to see the real type, e.g. if you hover the ConfigData from above you only see something like: `type ConfigData = ElementTaskEndPage & ElementTaskSwitchGateway & ElementSubProcessCall & ElementDatabase & ... 27 more ... & ElementSignalStartEvent`
with `Prettify<ConfigData>` you see something like `type ConfigData = {
    page: string;
    output: ScriptMapCode;
    case: WfCase;
    tasks: WfTask[];
    call: ScriptMapCode;
    processCall: string;
    cache: Cache;
    query: DbQuery;
    ... 34 more ...;
    attachToBusinessCase: boolean;
}`

not used currently, but really helpful if you try to write such ts types 🧙

Reply:
Wenn ich das richtig verstehe funktioniert das so:

Element1Data {
a:string,
b:string}
Element2Data {
c:string,
d:string}

ConfigData = {
a:string
b:string
c:string
d:string}

Element1InscriptionConfig {
a:string}
Element2InscriptionConfig {
c:string}

Man nimmt also alle Typen vom Protocol mixt die zusammen damit man einen Typ mit allen Properties hat und man nacher wieder einzelne Typen für die Inscription Masken erstellen kann die jedoch einzelne Properties aus dem Ursprünglichen Typen nicht mehr drin haben?

Warum macht man nicht für jeden Type eine Mapping Funktion die Protocol Objekt auf Inscription Object mapped anstelle hier mit Type Magic zu arbeiten, die nur wirkliche TS Experten verstehen?

Vielleicht hab ich auch einfach etwas nicht richtig verstanden.